### PR TITLE
Refactor packaging to account for IN_DOCKER.

### PR DESCRIPTION
**Pull request by jpoler**
@krobertson @wallyqs 

I had to refactor the way packages are being build because of some complications with the way we handle CI. Since all build steps are run within docker, mounting volumes was broken with my previous attempt.The container needs to be able to talk to the docker daemon running on its host, but volumes are mounted with respect to the host filesystem so the container really has no way of knowing what the absolute file path to a file in its local filesystem is.

The solution that we chose leverages the fact that the kurma repository is actually already mounted as a volume in the container running each build step. The build step container treats itself as a data volume container and refers to itself with the `--volumes-from` flag when calling docker run.

### CODE REVIEW

File: build/docker/deb/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -x
```
Comment (krobertson) [inline build/docker/deb/entrypoint.sh]: Probably want `set -e` so it errors out on a failure too.

File: build/docker/rpm/make-rpm-package.sh [hunk 2]
```diff
@@ -1,6 +1,8 @@
-#!/bin/sh
+#!/bin/bash
 
-PACKAGE_DIR=$(mktemp -d)
+set -x
```
Comment (krobertson) [inline build/docker/rpm/make-rpm-package.sh]: suggest `set -e` here too

File: build/docker/rpm/make-rpm-package.sh [hunk 3]
```diff
@@ -21,13 +23,22 @@ cp ./bin/kurma-api.aci \
...
+
+docker pull kurma/debian-fpm:latest
+docker pull kurma/centos-fpm:latest
+
+if [ -n "$IN_DOCKER" ]; then
+	docker run --rm \
+		   --volumes-from $HOSTNAME \
+		   -e KURMA_DIR=$(pwd) \
+		   -e VERSION=$VERSION \
+		   kurma/centos-fpm:latest
```
Comment (krobertson) [inline build/docker/rpm/make-rpm-package.sh]: Does this one need `TARGET_INIT` set like it does in the else?

File: build/docker/deb/entrypoint.sh (hunk 1 above)
Comment (jpoler) [inline build/docker/deb/entrypoint.sh]: fixed.

File: build/docker/rpm/make-rpm-package.sh (hunk 3 above)
Comment (jpoler) [inline build/docker/rpm/make-rpm-package.sh]: Whoops, it actually wasn't supposed to be in the else case.

**Comment by krobertson** [comment]
LGTM on greens.

**Pull request merged**